### DISCUSSION TITLE
upload and attach scans of documents

### DISCUSSION THREAD

**Discussion by sbts** [ledgersmb/LedgerSMB#5842] [Ideas]
as raised by k-man on #ledgersmb and expanded on by myself.

We should allow scans of document originals to be uploaded and viewed.
After creating a new document (eg: Vendor Invoice) and selecting an entity
- one or more documents should be able to be scanned and attached, eg: paper copy of the invoice, payment receipt.
- It should be possible to do this before any data other than the entity is entered.
- An entity should be *required* before upload is possible (this prevents documents from being created with no entity associated) (this could also be resolved if a document can't be saved without an entity instead)
- There should be a custom script name configurable for each entity that specifies post processing that should be run on image upload. This could be used to
  - resolve data from a QR code
  - OCR an image
  - pass the resulting data (QR or OCR) through a formatting script to be returned to pre-populate the document

This issue should be treated as multiple sub issues, with the first to be implemented being upload of an image provided by a client

**Comment by sbts** [comment]
as per http://ledgersmb.org/roadmap hasorli points out in the addons section there is 
> Photo or scan --> OCR for document storage (invoices, receipts) and retrieval
http://www.anix.no/p/OCR-2-ledgerSMB.txt (planning)

**Comment by jasonblewis** [comment]
My idea is that there should be a rudimentary document processing function in LedgerSMB. Imagine this typical scenario:

User has 50 receipts process. User scans the receipts and uploads them to LedgerSMB. Then user sorts them into various categories, for example inventory and non inventory receipts. User can then work through each receipt creating the necessary entity in LedgerSMB for that document. The point is there should be a way the user can batch process physical documents into LedgerSMB

**Comment by hasorli** [comment]
jasonblewis: The "batch process" has always been part of the original plan [1] .  (" Spool handler interface in LedgerSMB")  Thank for your support of LedgerSMB and "collect the documentation plan" . Lets make it happen :-) 

[1] http://www.anix.no/p/OCR-2-ledgerSMB.txt

**Comment by ylavoie** [comment]
A recent discussion showed interest  to have displayed barcode and/or images for inventory management too.

**Comment by ehuelsmann** [comment]
This issue has been superseded, at least in Europe, by ZugFerd: eInvoicing
